website/content/docs/overview/architecture/index.md

```diff
@@ -9,25 +9,57 @@ weight: 10
+![Architecture](workflow_multicluster.png)
 
-{{< bullet n=2 >}} A data streward configures policies in an external policy manager over assets defined in an external data catalog. The policy manager and data catalog are connected to {{< name >}} `PolicyCompiler` and `DataCatalog` services via `PolicyManagerAdapter` and `DataCatalogAdapter`, respectively.
+Before the data user can perform any actions a data operator has to [install](../../setup/quickstart) the {{< name >}} and modules
+that can be used by the user. [Modules](../modules) are a way that {{< name >}} abstracts data access for
+users. Modules can provide read/write access or produce implicit copies that serve as lower latency caches of
+remote assets. Modules also enforce policies defined for data assets. A detailed description of them can be found [here](../modules).
 
-{{< bullet n=3 >}} A developer submits a `M4DApplication` resource (e.g., via CI/CD pipeline) holding metadata about the application. such metadata includes the data assets required by the application and the processing purpose.
+{{< name >}} connects to external services to retrieve information about policies, available data and credentials.
+Policies, assets and access credentials to the assets have to be defined before the user can create an application. The
+current abstraction knows 3 different connectors but depending on the implementation the underlying system
+could be the same (e.g. a data catalog could store asset definitions and credentials). More information about connectors
+can be found [here](../connectors). It's designed in an open way so that multiple different catalog and policy frameworks
+of all kind of cloud and on-prem systems can be supported. The data steward configures policies in an external policy manager
+over assets defined in an external data catalog.
 
-{{< bullet n=4 >}} The _pilot_, a core component of {{< name >}}, processes the submission and retrieves a set of all governance actions required to be enforced according to policies. This is also performed upon any policy change.
+Once a developer submits a `M4DApplication` CRD to Kubernetes the ApplicationController will make sure that all
+the specs are fulfilled and will make sure that the data is made accessible to the user according to the
+previously defined policies. The `M4DApplication` holds metadata about the application such the data assets required by the 
+application, the processing purpose and the method of access the user wishes (protocol e.g. S3 or Arrow flight). 
+It uses this information to check with external systems (4) if access is allowed
+and whether restrictive policies such as masking or hashing have to be applied. It compiles blueprints depending
+on the policies defined by the connectors and chooses which modules (5) are best fit for the requirements that the user 
```

Comment:
on the **policy decisions received via** the connectors and chooses **the** modules (5) **which** are ...